### DISCUSSION TITLE
Obtaining historical data for expired futures contracts

### DISCUSSION THREAD

**Discussion by researchonly** [bmoscon/cryptofeed#780] [Q&A] [answered]
Is there any way to get the historical data for expired futures contracts, now I can get the non-expired contracts data through
`BinanceFutures().candles_sync('BTC-USDT-22H25', start='2022-01-01 00:00:00', end='2022-02-01 00:00:00', interval='1m')`
but when I tried with "BTC-USDT-21M25" it says "BTC-USDT-21M25 is not supported on BINANCE_FUTURES".

But the url format for the data retrieval is the same, so I wonder if there is a function that can get the historical data for expired future contracts?

Thanks

**Comment by bmoscon** [comment, marked as answer]
seems like it could be doable, would probably need to add a flag to methods that can return historical data to disable the symbol name checking. The issue is, if its expired, there is no way to generate a valid symbol mapping unless the symbols endpoint on the exchange provides the symbol name post expiration.

**Reply by researchonly** [reply]
Seems like a good idea to implement in the future, thanks for the reply.
